api/client/client_test.go

```diff
@@ -136,6 +137,50 @@ func TestClient_NewCollection(t *testing.T) {
 	})
 }
 
+func TestIt(t *testing.T) {
+	type Owner struct {
+		ID       core.InstanceID `json:"_id"`
+		ThreadID thread.ID       `json:"threadId"`
+	}
+	collectionName := "owners"
+
+	client, done := setup(t)
+	defer done()
+
+	id := thread.NewIDV1(thread.Raw, 32)
+	err := client.NewDB(context.Background(), id)
+	checkErr(t, err)
+	sch := util.SchemaFromInstance(Owner{}, false, true)
+	schBytes, err := json.MarshalIndent(sch, "", "  ")
+	checkErr(t, err)
+	schStr := string(schBytes)
+	t.Logf("\n%v", schStr)
```

Comment:
Ok here's the current output of the schema. The compromise/workaround here is that the schema must specify `additionalProperties=true` for any property that is a `thread.ID` or it must specify the `str` property inside a `properties` object. The library we use to create json schemas from an instance using reflection makes it easy to make `additionalProperties=true` for all properties, so that's what I'm doing for now. I'd like to find out of there's some way to force the presence of the `properties` and `str` property definition or at least only specify `additionalProperties=true` just for `thread.ID` properties, but it wasn't clear if that is possible with the lib we use.
```
        {
          "$schema": "http://json-schema.org/draft-04/schema#",
          "$ref": "#/definitions/Owner",
          "definitions": {
            "ID": {
              "additionalProperties": true,
              "type": "object"
            },
            "Owner": {
              "required": [
                "_id",
                "threadId"
              ],
              "properties": {
                "_id": {
                  "type": "string"
                },
                "threadId": {
                  "$schema": "http://json-schema.org/draft-04/schema#",
                  "$ref": "#/definitions/ID"
                }
              },
              "additionalProperties": true,
              "type": "object"
            }
          }
        }
```

Reply:
Hmm, I think I'm still leaning toward the string approach. This extra cruft doesn't buy us validation or anything like that, and makes the resulting JSON a bit weird, ie, `{"str": ID}`... that's a thread ID? Seems confusing to me.

Reply:
Ok cool, just wanted to explore that to be sure. String it is.

Reply:
Yeah totally, good to know how that's all working.